### PR TITLE
@-Variables for build-dirs, bugfix and minor refactor

### PR DESCRIPTION
**Pull request by duylethanh**
This PR contains the following:

- Added support for `@THIS_COMPILE_DIR@`, `@SOURCE_DIR_FOR:<build_name>@`, `@SOURCE_DIR_FOR:<build_name>@` and `@SOURCE_DIR_FOR:<build_name>@` variable in build args
- minor refactoring of string matching in `substitute()` in `invoke_run()`
- fix for a bug where `'_none'` was displayed as revision in the terminal for experiments that do not have a revision
- remove usage of Revision objects that had `'_none'` as a name by adapting the sorting key that is passed into `Config._expand_matrix()`

### CODE REVIEW

File: simexpal/build.py [hunk 1]
```diff
@@ -89,6 +89,17 @@ def get_concurrency():
 			n = os.cpu_count()
 		return n
 
+	def get_source_dir_for(build_name):
+		if build_name == build.name:
+			if not build.revision.is_dev_build:
+				return build.clone_dir
+			return build.source_dir
+
+		other_build = cfg.get_build(build_name, build.revision)
```
Comment (avdgrinten) [inline simexpal/build.py:98]: Why do we special case the current build here?

File: simexpal/build.py [hunk 2]
```diff
@@ -100,6 +100,13 @@ def get_source_dir_for(build_name):
 			return other_build.clone_dir
 		return other_build.source_dir
 
+	def get_compile_dir_for(build_name):
+		if build_name == build.name:
+			return build.compile_dir
+
+		other_build = cfg.get_build(build_name, build.revision)
```
Comment (avdgrinten) [inline simexpal/build.py:107]: Same here.

File: simexpal/build.py [hunk 3]
```diff
@@ -107,6 +107,13 @@ def get_compile_dir_for(build_name):
 		other_build = cfg.get_build(build_name, build.revision)
 		return other_build.compile_dir
 
+	def get_prefix_dir_for(build_name):
+		if build_name == build.name:
+			return build.prefix_dir
+
+		other_build = cfg.get_build(build_name, build.revision)
```
Comment (avdgrinten) [inline simexpal/build.py:114]: And here.

File: simexpal/build.py (hunk 1 above)
Comment (duylethanh) [reply simexpal/build.py:98]: It saves us a call of `cfg.get_build()`, which loops over all builds and revision. I can remove this case if you think it is overkill

Comment (avdgrinten) [reply simexpal/build.py:98]: That sounds more like an efficiency problem in `cfg.get_build()` to me. If this turns out to be too slow, we could always add a `dict()` of all builds.

Yes, please remove this special case here to simplify the code.

Comment (duylethanh) [reply simexpal/build.py]: Done.

**Comment by duylethanh** [comment]
I made the changes. The PR is ready to be merged.

**Comment by avdgrinten** [comment]
Thank you for the PR.

**Pull request merged**